### PR TITLE
Upgrade 0.16.1

### PR DESCRIPTION
**Pull request by HashEngineering**
For review only

### CODE REVIEW

**Comment by gruve-p** [comment]
Please remove:

file: https://github.com/Groestlcoin/groestlcoinj/blob/upgrade-0.16.1/README
directory: https://github.com/Groestlcoin/groestlcoinj/tree/upgrade-0.16.1/orchid
directory: https://github.com/Groestlcoin/groestlcoinj/tree/upgrade-0.16.1/misc

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@v1
+        with:
+          gradle-version: ${{ matrix.gradle }}
+          arguments: build --stacktrace
+      - name: Upload Test Results and Reports
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: bitcoinj-core-test-results-jdk${{ matrix.java }}-${{ matrix.os }}
```
Comment (gruve-p) [inline .github/workflows/gradle.yml]:
```suggestion
          name: groestlcoinj-core-test-results-jdk${{ matrix.java }}-${{ matrix.os }}
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-[![Build Status](https://travis-ci.org/bitcoinj/bitcoinj.png?branch=master)](https://travis-ci.org/bitcoinj/bitcoinj)   [![Coverage Status](https://coveralls.io/repos/bitcoinj/bitcoinj/badge.png?branch=master)](https://coveralls.io/r/bitcoinj/bitcoinj?branch=master) 
+[![Github Build Status](https://github.com/bitcoinj/bitcoinj/workflows/Java%20CI/badge.svg)](https://github.com/bitcoinj/bitcoinj/actions)
```
Comment (gruve-p) [inline README.md]:
```suggestion
[![Github Build Status](https://github.com/Groestlcoin/groestlcoinj/workflows/Java%20CI/badge.svg)](https://github.com/Groestlcoin/groestlcoinj/actions)
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
...
+[![GitLab Build Status](https://gitlab.com/bitcoinj/bitcoinj/badges/master/pipeline.svg)](https://gitlab.com/bitcoinj/bitcoinj/pipelines)
+[![Coverage Status](https://coveralls.io/repos/bitcoinj/bitcoinj/badge.png?branch=master)](https://coveralls.io/r/bitcoinj/bitcoinj?branch=master)
```
Comment (gruve-p) [inline README.md]:
```suggestion
[![GitLab Build Status](https://gitlab.com/Groestlcoin/groestlcoinj/badges/master/pipeline.svg)](https://gitlab.com/Groestlcoin/groestlcoinj/pipelines)
[![Coverage Status](https://coveralls.io/repos/Groestlcoin/groestlcoinj/badge.png?branch=master)](https://coveralls.io/r/Groestlcoin/groestlcoinj?branch=master)
```

File: README.md [hunk 4]
```diff
@@ -1,4 +1,6 @@
...
 
 [![Visit our IRC channel](https://kiwiirc.com/buttons/irc.freenode.net/bitcoinj.png)](https://kiwiirc.com/client/irc.freenode.net/bitcoinj)
```
Comment (gruve-p) [inline README.md]:
```suggestion
[![Visit our IRC channel](https://kiwiirc.com/nextclient/irc.libera.chat/groestlcoin.png)](https://kiwiirc.com/nextclient/irc.libera.chat/groestlcoin)
```

File: README.md [hunk 5]
```diff
@@ -37,6 +39,40 @@ gradle clean assemble
 
 Alternatively, just import the project using your IDE. [IntelliJ](http://www.jetbrains.com/idea/download/) has Gradle integration built-in and has a free Community Edition. Simply use `File | New | Project from Existing Sources` and locate the `build.gradle` in the root of the cloned project source tree.
 
+### Building and Using the Wallet Tool
+
+The **bitcoinj** `tools` subproject includes a command-line Wallet Tool (`wallet-tool`) that can be used to create and manage **bitcoinj**-based wallets (both the HD keychain and SPV blockchain state.) Using `wallet-tool` on Bitcoin's test net is a great way to learn about Bitcoin and **bitcoinj**.
+
```
Comment (gruve-p) [inline README.md]:
```suggestion
The **groestlcoinj** `tools` subproject includes a command-line Wallet Tool (`wallet-tool`) that can be used to create and manage **groestlcoinj**-based wallets (both the HD keychain and SPV blockchain state.) Using `wallet-tool` on Groestlcoin's test net is a great way to learn about Groestlcoin and **groestlcoinj**.

```

**Comment by gruve-p** [comment]
Can you please replace (where possible) bitcoinj with groestlcoinj on https://github.com/Groestlcoin/groestlcoinj/blob/upgrade-0.16.1/README.md

**Comment by gruve-p** [comment]
we no longer have a alert key, so these can be removed:

https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/org/bitcoinj/core/CoinDefinition.java#L244-L245

**Comment by gruve-p** [comment]
Cryptsy no longer exists and can be removed:
https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/org/bitcoinj/core/CoinDefinition.java#L22-L23

**Comment by gruve-p** [comment]
Can we add regtest and signet support:
https://github.com/bitcoinj/bitcoinj/commit/6ad636cceca1c6b52dde37ab98ab1999ad18c8ed
https://github.com/bitcoinj/bitcoinj/commit/57bbb9152b0e60c010e657712293ed90407a9f8f

**Comment by gruve-p** [comment]
Bump protocol version to 70013:
https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/org/bitcoinj/core/CoinDefinition.java#L51

**Comment by gruve-p** [comment]
I think its cleaner to change https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/org/bitcoinj/core/CoinDefinition.java#L37 into:
public static final int TARGET_SPACING = 60;  // 60 seconds per block.

**Comment by gruve-p** [comment]
I think its cleaner to change [nPowTargetTimespan](https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/org/bitcoinj/core/CoinDefinition.java#L36) to:
public static final int TARGET_TIMESPAN = 14 * 24 * 60 * 60;  // 1 day per difficulty cycle, on average.

**Comment by gruve-p** [comment]
Shouldnt [this](https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/org/bitcoinj/core/CoinDefinition.java#L52) be bumped to [31800](https://github.com/Groestlcoin/groestlcoin/blob/master/src/version.h#L18)

**Comment by gruve-p** [comment]
Can we remove [this](https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/org/bitcoinj/core/CoinDefinition.java#L108-L113) as per https://github.com/bitcoinj/bitcoinj/pull/2279

**Comment by gruve-p** [comment]
Do we need: https://github.com/Groestlcoin/groestlcoinj/blob/master/core/src/main/java/com/hashengineering/crypto/X11.java

**Comment by gruve-p** [comment]
Do we need all the hashes in:
https://github.com/Groestlcoin/groestlcoinj/tree/master/core/src/main/java/fr/cryptohash

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (HashEngineering) [reply .github/workflows/gradle.yml]: `bitcoinj-*`  was added in many places in version 0.16, so we have to rename all of them to `groestlcoinj-*`

File: README.md (hunk 3 above)
Comment (HashEngineering) [reply README.md]: We are not using GitLab for anything, we could remove it  -- or we could also publish to GitLab.  Also, we don't use coveralls.io, but we could add support for it.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (gruve-p) [reply .github/workflows/gradle.yml]: This is just the name of the ci task

File: README.md (hunk 3 above)
Comment (gruve-p) [reply README.md]: In that case lets remove Gitlab and coveralls

**Comment by gruve-p** [comment]
Please remove:
https://github.com/Groestlcoin/groestlcoinj/blob/upgrade-0.16.1/core/src/main/resources/org.bitcoin.test.checkpoints
https://github.com/Groestlcoin/groestlcoinj/blob/upgrade-0.16.1/core/src/main/resources/org.groestlcoin.production.checkpoints

Please rename https://github.com/Groestlcoin/groestlcoinj/blob/upgrade-0.16.1/core/src/main/resources/org.bitcoin.test.checkpoints.txt to https://github.com/Groestlcoin/groestlcoinj/blob/upgrade-0.16.1/core/src/main/resources/org.groestlcoin.test.checkpoints.txt

Please create https://github.com/Groestlcoin/groestlcoinj/blob/upgrade-0.16.1/core/src/main/resources/org.groestlcoin.signet.checkpoints.txt

File: core/build.gradle [hunk 6]
```diff
@@ -123,4 +160,4 @@ uploadArchives {
             }
         }
     }
-}
+}*/
```
Comment (gruve-p) [inline core/build.gradle]: Do we need this in?

**Comment by HashEngineering** [comment]
> Do we need all the hashes in: https://github.com/Groestlcoin/groestlcoinj/tree/master/core/src/main/java/fr/cryptohash

we don't, in fact we can import a library that has all this stuff and can remove all of these files.

File: core/build.gradle (hunk 6 above)
Comment (HashEngineering) [reply core/build.gradle]: yes, though we may change it to a different maven plugin for publishing.

Comment (HashEngineering) [reply core/build.gradle]: this is the command that publishes to maven central.

Comment (HashEngineering) [reply core/build.gradle]: it is currently commented out because it doesn't "work" or compile after the upgrade to 16.1

**Pull request merged**